libs/infinity_emb/infinity_emb/log_handler.py

```diff
@@ -1,21 +1,218 @@
 # SPDX-License-Identifier: MIT
 # Copyright (c) 2023-now michaelfeil
 
+import json
 import logging
-import sys
+import re
+from contextvars import ContextVar
 from enum import Enum
-from typing import Any
+from functools import wraps
+from time import perf_counter, time
+from traceback import format_exception
+from uuid import uuid4
 
-logging.getLogger().handlers.clear()
+from fastapi import Request, HTTPException
+from opentelemetry.instrumentation.logging import LoggingInstrumentor
+from starlette.middleware.base import BaseHTTPMiddleware
+from starlette.responses import StreamingResponse
 
-handlers: list[Any] = []
-try:
-    from rich.console import Console
-    from rich.logging import RichHandler
 
-    handlers.append(RichHandler(console=Console(stderr=True), show_time=False))
-except ImportError:
-    handlers.append(logging.StreamHandler(sys.stderr))
+class StructuredLogging:
+    request_time = ContextVar("request_time", default=perf_counter())
+    account_id = ContextVar("account_id", default="null")
+    request_id = ContextVar("request_id", default="No request ID")
+    trace_parent = ContextVar("trace_parent", default="-")
+
+    _config_dict = {
+        "ts": "%(timestamp_ms)d",
+        "type": "app",
+        "svc": "freddy-infinity",
+        "lvl": "%(levelname)s",
+        "act": "%(pathname)s:%(funcName)s:%(lineno)d",
```

Comment:
This will print the pathname and funcName in the place this log is getting printed.
Can we have the funcName of the function which is executing?

Reply:
Done